### ISSUE TITLE
It can be submitted even if required fields are blank.

### ISSUE THREAD

**Issue by Toraji123** [pods-framework/pods#6352]
### Description

I have created a form like this.
```
$fields = array('name' => array( 'required' => 1 ));
echo $mypod->form( $fields, 'Save' );
```
This is how it will appear on the front end.
![image](https://user-images.githubusercontent.com/40485661/144732678-7e55e801-a031-4d86-b4df-f089fe217400.png)

It can be submitted even if this required field is blank.

Since it is a required field, can you please make it impossible to submit it?

Up to v2.7.31, an error message was displayed and it could not be submitted.
![image](https://user-images.githubusercontent.com/40485661/144732793-c98cf025-5775-466d-af2b-4915e80a84fc.png)

Thanks in advance.

### Version

2.8.7

**Comment by JoryHogeveen** [comment]
Another report: https://wordpress.org/support/topic/probleme-when-i-save-a-custom-post-with-empty-field-required/

**Comment by Quatrys** [comment]
Hello For me this error is in admin page on custom content type

**Comment by HmCody** [comment]
I am also running into this problem. And what makes it worse is that the publish triggers pods_api_pre_save_pod_item and there is no indication that the required fields are not present short of testing them all.

**Comment by HmCody** [comment]
One more detail. If a required field has a default value, that value is substituted for the blank field without being sent to error processing. Only required fields that do NOT have a default value now throw an error.

**Comment by Quatrys** [comment]
I have partialy sove the problem with this function in JS : 
But only working on field inout[text] : 

function pods_admin_submit_validation(valid_form, $submittable ){

    let erreur=0;
    let champs='';
    $submittable.find('.pods-validate-required').each(function( index ) {
        let tr_pods_field__container_field=jQuery( this ).parent('th').parent('tr.pods-field__container').find('.pods-dfv-container input.pods-form-ui-field').val();

        if(tr_pods_field__container_field != undefined){
            console.log(tr_pods_field__container_field);
        let tr_pods_field__container_field_label=jQuery( this ).parent('th').parent('tr.pods-field__container').find('label.pods-validate-required').text();
        if(!tr_pods_field__container_field){
            jQuery( this ).parent('th').parent('tr.pods-field__container').find('.pods-dfv-container input.pods-form-ui-field').trigger('focus');
            champs+=tr_pods_field__container_field_label;
            erreur++;
        } }
      })

      if(erreur>1){return false;}else{return true;}
      //return false;
  
    
}

**Comment by sc0ttkclark** [comment]
@zrothauser we need a way to detect if the fields have met their requirements when handling submitting through our public forms JS. I know we'll be revisiting the other screens in other work but I wonder if there's some sort of API or function we can expose here to detect if the fields for a form are OK.

**Comment by Toraji123** [comment]
Do you have any plans to fix this bug?
I and my customers will be very happy if this bug will be fixed.

**Comment by sc0ttkclark** [comment]
We need to go through this and determine what's left and what is already covered by the work planned for #2424.

Some of the problems described are resolved now but I just want to do a comprehensive review of the additional problems added on to this issue so we can figure out whether those should be their own separate issues or if we should solve them as part of this one.

**Comment by Toraji123** [comment]
@sc0ttkclark 
It appears that this issue has not yet been resolved. It would be very helpful if you could fix it.

**Comment by sc0ttkclark** [comment]
Fixed and tested in 3eca8caa196110a65766a5c6df9dc87e6f18e3fa

Thanks everyone for your patience here, there's still more to do later for other validation rules and to make that all more dynamic with the DFV JS involved but it now properly picks up the required validation and will stop itself from submitting.
